### PR TITLE
Migrate some field mapper tests to ESTestCase

### PR DESCRIPTION
**Pull request by nik9000**
This switches a few tests for field mappers from `ESSingleNodeTestCase`
to `ESTestCase` because, in general, we prefer to avoid
`ESSingleNodeTestCase` when we can because it is slow and "big". "Big"
here means that it pulls in an entire node, making it difficult to
reason about what you are testing.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/index/mapper/FieldMapperTestCase2.java [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+import java.util.function.BiConsumer;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+
+/**
+ * Base class for testing {@link FieldMapper}s.
+ * @param <T> builder for the mapper to test
+ */
+public abstract class FieldMapperTestCase2<T extends FieldMapper.Builder<?>> extends MapperTestCase {
```
Comment (nik9000) [inline test/framework/src/main/java/org/elasticsearch/index/mapper/FieldMapperTestCase2.java:47]: I'm hoping that we'll migrate to this one and remove the `2` in the name.

Comment (nik9000) [reply test/framework/src/main/java/org/elasticsearch/index/mapper/FieldMapperTestCase2.java:47]: Or we'll drop this entirely when we move to `ParameterizedFieldMapper`. Tests of those depend on `MapperTestCase` directly.

File: test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            Map.of("default", new NamedAnalyzer("default", AnalyzerScope.INDEX, new StandardAnalyzer())),
+            Map.of(),
+            Map.of()
+        );
+    }
+
+    /**
+     * Create the {@link MapperService} for an index.
+     */
+    protected final MapperService createIndex(Settings settings, XContentBuilder mapping) throws IOException {
```
Comment (nik9000) [inline test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java:88]: `MapperService` seems good enough for most of the tests. Most tests actually just want `DocumentMapper` which it contains. `MapperService` is useful for testing merging mappings though.

Comment (romseygeek) [reply test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java:88]: Can we call this `createMapperService` instead?  And maybe add a `createDocumentMapper` sugar method as well, given that the vast majority of callers immediately call `documentMapper()`?

**Review by romseygeek — APPROVED** [review]
I left a comment about method naming, but otherwise this looks great!

I also wonder if it's worth adding a method that checks that the minimal mapping serializes to itself?  There are some exceptions (CompletionFieldMapper I'm looking at you) but I think we should enforce this for all new mapper implementations at least, and base classes like this are good ways to check behaviour.

**Comment by nik9000** [comment]
@elasticmachine update branch

**Comment by nik9000** [comment]
> I also wonder if it's worth adding a method that checks that the minimal mapping serializes to itself? There are some exceptions (CompletionFieldMapper I'm looking at you) but I think we should enforce this for all new mapper implementations at least, and base classes like this are good ways to check behaviour.

I forget to get this before merging so I'll grab this is a follow up change.

**Pull request merged**